plrust/src/prosrc.rs

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2021-2023 Technology Concepts & Design, Inc. <support@tcdi.com>
+
+All rights reserved.
+
+Use of this source code is governed by the PostgreSQL license that can be found in the LICENSE.md file.
+*/
+
+//! Routines for managing the `pg_catalog.pg_proc.prosrc` entry for plrust functions
+use std::collections::BTreeMap;
+use std::rc::Rc;
+
+use pgx::pg_sys;
+use pgx::pg_sys::MyDatabaseId;
+use pgx::prelude::PgHeapTuple;
+use serde::{Deserialize, Serialize};
+
+use crate::error::PlRustError;
+use crate::pgproc::PgProc;
+use crate::target;
+use crate::target::CompilationTarget;
+use crate::user_crate::{FnReady, UserCrate};
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+struct ProSrcEntry {
+    src: String,
+    lib: BTreeMap<String, Vec<u8>>,
```

Comment:
Wait. What do each of these Strings represent? Isn't one a compile target? Why not just use that type?

Reply:
Oh. Sure. Duh. Can make CompilationTarget derive the serde traits too. Will fix